### ISSUE TITLE
Parmeters set with in: header are not added to headers

### ISSUE THREAD

**Issue by FinnIckler** [drwpow/openapi-typescript#1230]
**Description**

When you have a parameter in your openAPI spec with in: header, it is not added to the headers in the request

**Reproduction**

openapi spec like this:
```
(...)
paths:
  "/api/v1/register":
    post:
      summary: Add an attendee registration
      parameters:
      - name: SomeHeader
        in: header
        schema:
          type: string
      responses:
        '202':
          description: only required fields included
(...)
```
Generates a schema file like this
```
(...)
  "/api/v1/register": {
    /** Add an attendee registration */
    post: {
      parameters: {
        header?: {
          SomeHeader?: string;
        };
      };
(...)
```
which I use like this:
```
const { data, error, response } = await post('/api/v1/register', {
    params: { header: { SomeHeader: value } },
    body,
  })
```
If I look into the request in Firefox, the header parameter is not send with it. I also took a quick look at the code and I don't think this is in there either.
**Expected result**
Header to be added to the headers. 

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

**Comment by drwpow** [comment]
ah good call out—I think this is indeed missing.

🤔 Just from an API perspective, we do have the `params` key to namespace it from other options like generic (untyped) `headers` and `body`. But would it be weird to have both `headers` and `params.headers`?

```js
get('/some/url', {
  headers: {
    foo: 'bar',
  },
  params: {
    headers: {
      typedHeader: 'value',
    },
  },
});
```

My initial take is “yes,” because probably this is rare—headers will usually be defined on `createClient()` and not on the method calls. Plus, there would be a distinction between typed (in your OpenAPI schema) and untyped headers (not in your schema). But wanted to gauge others’ knee-jerk reaction (after all, a key goal of this project is a clean, easy-to-use API 🙂)

**Comment by angularsen** [comment]
+1, just stumbled on this on my initial experimentation with this library

**Comment by drwpow** [comment]
Fixed in `0.6.2`!
